sklearn_numba_dpex/kmeans/kernels/compute_labels.py

```diff
@@ -80,9 +85,12 @@ def assignment(
         assignments_idx,          # OUT            (n_samples,)
     ):
         # fmt: on
-
-        sample_idx = dpex.get_global_id(zero_idx)
-        local_work_id = dpex.get_local_id(zero_idx)
+        local_row_idx = dpex.get_local_id(zero_idx)
+        local_col_idx = dpex.get_local_id(one_idx)
+        sample_idx = (
+            (dpex.get_global_id(zero_idx) * sub_group_size)
+            + local_col_idx
+        )
```

Comment:
I admit I am lost here. Could you add some comments to explain how `sub_group_size` and the 2d local indexing logic work?

Reply:
Will add a dosctring in the `lloyd` kernel (where most of the explanations for similar kernels are located)